File: 프로그래머스/lv2/42577. 전화번호 목록/전화번호 목록.java
import java.io.*;
import java.lang.*;
import java.util.*;
class Solution {
    public boolean solution(String[] phone_book) {
        Set<String> set=new HashSet<>();
        for(int i=0;i<phone_book.length;i++){
            set.add(phone_book[i]);
        }
        for(String p:phone_book)
            for(int i=1;i<p.length();i++){
                if(set.contains(p.substring(0,i))){
                    return false;
                }
            }
        
        return true;
    }
}

File: 프로그래머스/lv2/42577. 전화번호 목록/전화번호 목록.py
def solution(phone_book):
    
    phone_book.sort()##정렬
    for i in range(len(phone_book) - 1) :
        if phone_book[i+1].startswith(phone_book[i]) :
            return False ###인접한인덱스만비교
        
    return True